### PR TITLE
Change Auth method parameter to more strict and informing struct para…

### PR DESCRIPTION
**Pull request by ElecTwix**
Change Auth methods parameter to a more strict and informative struct parameter
few typo fixes

### CODE REVIEW

**Comment by phughk** [comment]
Could go mod tidy be separate please? I would like to compare the diff conveniently and revert if necessary 🙏

**Comment by ElecTwix** [comment]
> Could go mod tidy be separate please? I would like to compare the diff conveniently and revert if necessary 🙏

Done 👌🏼

File: README.md [hunk 1]
```diff
@@ -41,10 +41,14 @@ func main() {
 		panic(err)
 	}
 
-	if _, err = db.Signin(map[string]string{
-		"user": "root",
-		"pass": "root",
-	}); err != nil {
+
```
Comment (phughk) [inline README.md:44]:
```suggestion
```

**Review by phughk — APPROVED** [review]
Nice, thanks!

**Pull request merged**